src/com/xilinx/rapidwright/util/ParallelismTools.java

```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright (c) 2021 Xilinx, Inc.
+ * All rights reserved.
+ *
+ * Author: Eddie Hung, Xilinx Research Labs.
+ *
+ * This file is part of RapidWright.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.xilinx.rapidwright.util;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.RunnableFuture;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Utilities to aid in parallel processing
+ *
+ * A class that abstracts away single-threaded and multi-threaded execution.
+ * Single-threaded mode means that all tasks submitted will be executed
+ * immediately (on the submitting thread).
+ */
+public class ParallelismTools {
+    private static boolean parallel = false;
```

Comment:
Can we make parallel DCP writes on by default?

Reply:
I would suggest we turn it on at the next release once it's had some time to bed down and get tested externally.

Reply:
I think the next release sounds reasonable.  I also wonder if we should provide a separate mechanism (environment variable, etc) that would allow users to change this setting without having to write code/recompile, etc.